### ISSUE TITLE
Connection errors when using cron

### ISSUE THREAD

**Issue by pzystorm** [hypery2k/owncloud#296]
My roundcube is only available via https://mydomain.de/roundcube, not via http://...
When the cron runs every 15 minutes, it correctly identifies my hostname, but in the file apps/roundcube/lib/RoundCubeLogin.class.php ...

>        private function sendRequest($path, $postData = false) {
>                if (is_array($postData)) {
>                        $postData = http_build_query($postData, '', '&');
>                }
>                $method = (!$postData) ? "GET" : "POST";
>                if ((isset($_SERVER['HTTPS']) && $_SERVER["HTTPS"] || isset($_SERVER['HTTP_X_FORWARDED_PROTO']) &&
>                                $_SERVER['HTTP_X_FORWARDED_PROTO'] == 'https')) {
>                        $url = "https://";
>                } else {
>                        $url = "http://";
>                }

.. all the $_SERVER variables are of course empty (when called by cron resp. console) and so it is tried to connect to roundcube via http://mydomain.de/roundcube/.... And this raises an error.

As ugly workaround I have changed the 'else' part and have replaced 'http' by 'https'.

**Comment by mckaygerhard** [comment]
i question, the owncloud are under http and the rouncube under https right?

**Comment by pzystorm** [comment]
Both under https, http is not possible in my apache configuration

**Comment by mckaygerhard** [comment]
seems that the server vars are not send in the headers.. 

please provide more info: are u using cgi or mod_php? what version of apache, version of php, apache own compiled ocnfiguration or distribution packages?

**Comment by pzystorm** [comment]
neither cgi nor mod_php
Just a normal cronjob, inserted via "crontab -u webserveruser -e"
The cron executes the following command:
*/15  *  *  *  * php -c /www/owncloud/conf /www/owncloud/web/cron.php
So my apache configuration doesnt matter in this case, because it is not part of the chain.
When u call a php script via console, there are only very few environment variables, and $_SERVER is not filled at all.

**Comment by mckaygerhard** [comment]
hi @pzystorm of course and seems that's the reason.. so u hav right about the apache configuration, but due in fact makes sense I do not understand relationship between the tasks of oc and login into the rc, i setup a similar env in my lighttpd and made the oc use cron jobs event page reload jobs, get that error only when my http access method are not available...

so if u use the cron job that make login taks, so then code must change to use others detection method, but i not sure this are not so easy due security reasons... 

i can made a workaroud but i waith a response opinion from @hypery2k

**Comment by hypery2k** [comment]
It's possible to add a second parameter which can be used within the cron, but what is the exact use case for your setup? What are you trying to achieve?

**Comment by pzystorm** [comment]
Ahh, ok, yes, thats true. my http access method is not available.
owncloud and roundcube are only available via https.

**Comment by mckaygerhard** [comment]
well @pzystorm of course may faill, u only have https, and code firts try normal htttp, due the env of servers are not detectec (run and call by curl), the code fails, due fallback to http that its not available..as i said, a workaround can be made, but this are rare and seems very specific to you..

so the question now its, for what cron? what its the relationship between cron and the oc&rc login?
we dont undestandt this strange/interesting setup!

if the configuration u parse are very specific to you, i think this its not a but and maybe, maybe could be a feature request.. but not so important to delegate to next milestone...

**Comment by hypery2k** [comment]
so why are you using the cron for login?

**Comment by pzystorm** [comment]
I also don't know. I have just setup this cron, because it is demanded by owncloud.
And since enabling this roundcube app in owncloud, I realise those errors in the log.
When disabling the app, there are no errors.
So my guess is, that this cron validates the roundcube session IDs... 
In the error log is "Failed to refresh the RC session", from OC_RoundCube_App.class.php->refresh()
Every 15 minutes, because my cron runs every 15 minutes.
No change whether someone is logged in.

**Comment by hypery2k** [comment]
so you're justing seeing the error in the log, but the login itself is working? rights?

**Comment by pzystorm** [comment]
Yep. Exactly.

**Comment by mckaygerhard** [comment]
well i'm confusing now? plguin are working and not working!!! jajajaja

of course when cron ask for the url there's no login and a error will raise. .but plugin its working for normal behavior!

**Comment by pzystorm** [comment]
Sorry, guys, but are you stupid? And sorry for filling this bug... will never do this again...

I am not the developper... this are you... and so you just should know, why this damn cronjob wants to validate each time the cron is executes a roundcube session.
The plugin is not completely working. After using owncloud a bit (maybe one hour or so) I cannot switch to the roundcube app anymore because I am logged out there.

> Es bestehen Konfigurationsfehler
> Bitte korrigieren sie die Administrationseinstellungen
> Einstellungen
> ERROR: Technical problem, Unable to login to roundcube 

When logging in and switching to the roundcube app directly after a few minutes, everything works.

And my hope is, that this cronjon would keep this session alive... 

Does nobody use a cronjob combined with a https-only-environment??? Everyone using this setup should encounter this bug.

**Comment by mckaygerhard** [comment]
ok calm down... please can u enable more debug log and parse the error debug (and in english please)

**Comment by hypery2k** [comment]
first of all, no need to get personally! It's an open source project which is available for free. So please, don't blame anybody!

You never really stated out what you expected. That would be helpful. You just told us, you get an log message...

The cronjob was a legacy functionality, that wasn't needed anymore. Within OwnCloud for the original purpose of refreshing the routes are used!. So the functionality/feature is still in the app, just via another implementation.
My latest tests were working even without this legacy code

So long story short: This code wasn't really used and thats why it was removed to don't show this irritating log entry anymore.
